info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java

```diff
@@ -76,6 +88,41 @@ public InfoCompiler(Datastore datastore) throws IOException {
     this.datastore = datastore;
     this.webCrawler =
         new WebCrawler(this.datastore, new NewsContentExtractor(), new RelevancyChecker());
+    parseAddressesFromDataset();
+  }
+
+  /**
+   * Reads the National Address Database (Release 3), referenced below, from Google Cloud Storage.
+   * The file is located at bucket {@code Config.ADDRESS_BUCKET_NAME} and named {@code
+   * Config.ADDRESS_FILE_NAME}.
+   *
+   * @see <a href=
+   *    "https://www.transportation.gov/gis/national-address-database/national-address-database-0">
+   *    National Address Database (Release 3)</a>
+   */
+  private void parseAddressesFromDataset() {
+    Storage storage =
+        StorageOptions.newBuilder().setProjectId(Config.PROJECT_ID).build().getService();
+    String[] addressFile =
+        new String (storage.get(Config.ADDRESS_BUCKET_NAME, Config.ADDRESS_FILE_NAME,
+                                Storage.BlobGetOption.userProject(Config.PROJECT_ID))
+                        .getContent()).split("\\r?\\n");
```

Comment:
Was this indentation intentional?

Reply:
Yes. `storage.get(Config...., Config....., Storage.BlobGetOption......)` is a Blob. then `.getContent()` is applied to that Blob. So `storage.get(...) `is one layer, while `.getContent()` is one layer deeper (so 4 spaces indented).